lib/burnside/bridge.cc

```diff
@@ -458,9 +462,39 @@ class FIRConverter {
     build().create<M::CallOp>(toLocation(), callee, operands);
     build().create<fir::UnreachableOp>(toLocation());
   }
-  void genFIR(const Pa::ReturnStmt &stmt) {
-    build().create<M::ReturnOp>(toLocation());  // FIXME: argument(s)?
+  void genReturnStmt(AnalysisData &, const Pa::FunctionSubprogram &func) {
+    auto &stmt{std::get<Pa::Statement<Pa::FunctionStmt>>(func.t)};
+    auto &name{std::get<Pa::Name>(stmt.statement.t)};
+    assert(name.symbol);
+    const auto &details{name.symbol->get<Se::SubprogramDetails>()};
+    const Se::Symbol *result{&details.result()};
+    M::Value *resultRef{symbolMap.lookupSymbol(result)};
+    assert(resultRef);  // FIXME might die if result
+    // was never referenced before and temp not created.
```

Comment:
The FIR should return an `undefined` value if the `FUNCTION` doesn't actually return a well-defined value.

Reply:
Good point. I still need to think more here. I am not sure the fact that the symbol is not part of the symbolMap means it does not have a value. Maybe another variable could have been associated to the result and this variable defined. I think this should not be possible (for instance function result cannot appear in equivalence), but I have not surveyed all the ways this could happen.
If we cannot be sure enough, the safest would be to create the temp and leave llvm/mlir later decide if it is undefined or was indirectly define.